### DISCUSSION TITLE
Portal-App: Zoom auf die aktuelle Position

### DISCUSSION THREAD

**Discussion by thomaskofler** [e-netze/webgis-community#169] [General]
Hallo Jürgen,

bei uns verwenden die Flussmeister die Portal-App zur Datenerfassung von beschädigten Bäumen entlang Flussabschnitten, die dem Land gehören. Zur exakten Bestimmung der Position eines Baumes wird eine GPS-Antenne ans Tablet angeschlossen und mit der Funktion _Zoom auf die aktuelle Position_ ermittelt. Das funktioniert ziemlich gut, man muss aber dann den Punkt neben dem Standort-Symbol setzen und dann draufziehen (direkt draufsetzen geht nicht):

![grafik](https://github.com/gis-eni/webgis-forum/assets/119412172/1fc29dbe-2938-4b84-8ddd-4f59f43dfb23)

Jetzt war die Frage, ob es möglich wäre, ein Werkzeug zu machen, dass der Punkt automatisch gesetzt wird, wenn man die Funktion der Standortbestimmung auswählt.

**Comment by gis-eni** [comment]
Hallo Thomas,

das Tool gibt es eigentlich schon. Bei uns wird das schon genau so verwendet.
Wenn der Anwender etwas zeichnet, sieht auf link einen *Bubble* mit der Aufschrift ``GPS``:

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/651f3a20-8bd6-4d68-bf28-8518cb268e73)

Zieht man diese *Bubble* irgendwo hin in die Karte, wird die GPS Position ermittelt und es erscheint ein Fadenkreuz Symbol an der Position. Die Bubble wird dann grün und die aktuelle GPS Genauigkeit wird angezeigt (geht bei mir an Desktop jetzt leider nicht):

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/b0bc212c-3764-4b8d-b139-63d608e75f44)

Wenn du dann auf die grüne gewordene GPS Bubble klickst, wir der nächste Vertices an der aktuellen GPS Position gesetzt.
Dann kannst du theoretisch weiter gehen und dann wieder auf die Bubble Klicken um den nächsten Vertex zu setzen (Bei Linien).

Wenn du die Funktion wieder beenden willst, schiebst du die Bubble einfach wieder an den linken Rand/Mitte.

Damit das funktioniert musst du in der ``custom.js`` folgendes einstellen:

``` javascript
if (mapUrlName === "Baumkataster") {
    webgis.currentPosition = webgis.currentPosition_watch;

    webgis.currentPosition.minAcc = 5;   // [m]
    webgis.currentPosition.maxAgeSeconds = 0.1;  // [s]
    webgis.currentPosition.useWithSketchTool = true; 

    //webgis.continuousPosition.useTransformationService = true;
    //webgis.continuousPosition.transformationSpatialValidity = 20;

    //webgis.continuousPosition.helmert2d = {
    //    srs: 31256,
    //    Cx:  0.600,
    //    Cy: -0.234,
    //    Rx: -67946.151,
    //    Ry: 215079.498,
    //    r: 399.9992 * Math.PI / 200,
    //    scale: 1 + (-6.576 * 1e-6)
    //};
}
```

mit ``minAcc`` kannst du bestimmen, wie genau die GPS Position sein muss, damit die Bubble grün wird und der Punkt gesetzt werden kann.

Achtung: Wenn du Genauigkeiten im Dezimeter Bereiche haben willst, musst du eventuell lokale Transformation Parameter anwenden!

**Comment by thomaskofler** [comment]
Danke für die schnelle Antwort, funktioniert in der Karte super, in der Portal-App aber nicht oder? Was ok wäre, weil ich kann für die Anwendung auch eine Karte verwenden, da hätte ich noch eine Frage und zwar, erstellt man ein xml-Edit File kann man beim Attribut eine Checkbox mitgeben:
![grafik](https://github.com/gis-eni/webgis-forum/assets/119412172/dab005e5-1e0c-44f6-b46f-a116af8bb3c7)

Schaut dann in der Portal-App so aus:
![grafik](https://github.com/gis-eni/webgis-forum/assets/119412172/2692e98c-6b41-4938-b780-6afa154d332b)

In der Portal-App werden dann die Checkboxen angezeigt, im Editierbereich einer Karte aber nicht. Gibt es da eine Möglichkeit, dass im Editierbereich einer Karte auch die Checkboxen angezeigt werden? Und könnte man das auch über das CMS lösen, weil dann müsste ich kein xml erstellen und hätte alles im CMS beinander.

**Reply by thomaskofler** [reply]
Oder was optional auch gehen würde, ist es möglich ein Dropdown mit Mehrfachauswahl zu erstellen (in ArcGIS Pro geht es nämlich nicht)?

**Reply by gis-eni** [reply]
Da stellt sich für mich die Frage, wie das dann im Ziel gespeichert werden soll? Ein Edit Field entspricht in der Regel einem Feld in der Ziel Featureklasse...

Bei "Checkbox" stellt sich für mich jetzt auch die Frage, was hier der Zeil Feldtyp in der Datenbank sein sollte?

**Reply by thomaskofler** [reply]
Feldtyp ist einfach ein Text, man sieht eh in dem Code-Beispiel, dass wenn die Checkbox aktiviert ist, dann zB der Default-Value Gebäude gespeichert wird (ansonsten bleibt das Feld leer). Ich möchte es deshalb über eine Checkbox lösen, weil hier ausschließlich mit dem Smartphone oder Tablet editiert wird und anhaken halt schneller geht als etwas reinzuschreiben. Auswahlliste funktioniert da nicht, weil es soll eine Mehrfachauswahl möglich sein.

**Reply by gis-eni** [reply]
Besser wären hier Auswahllisten oder Datenmodell ändern.
